functions-v2/src/on-document-summarized.ts

```diff
@@ -0,0 +1,184 @@
+  aiAgreements: Record<string, AiAgreement>;
+}
+
+export interface AiAgreement {
+  version: 1;
+  value: string;
+  content: string;
+  tags: string[];
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const logInfo = (...message: any[]) => logger.info("ON DOCUMENT SUMMARIZED:", ...message);
+
+const openaiApiKey = defineSecret("OPENAI_API_KEY");
+
+/*
+ * onDocumentSummarized
+ * Listens for changes to comments and creates/updates the document summary along with its ai agreements.
+*/
+export const onDocumentSummarized = onDocumentWritten({
+  document: "{root}/{space}/documents/{documentId}/comments/{commentId}",
+  secrets: [openaiApiKey],
+}, async (event: FirestoreEvent<Change<DocumentSnapshot> | undefined>) => {
+  if (!event.data) return;
+
+  logInfo("START");
+
+  // ensure the comment either has a uid and has or had agreeWithAi set - we don't care about comments that don't
+  const before = event.data.before?.data();
+  const after = event.data.after?.data();
+  const commentUid = before?.uid ?? after?.uid;
+  if (!commentUid || !(before?.agreeWithAi || after?.agreeWithAi)) {
+    logInfo("EARLY EXIT: no uid or agreeWithAi not set:", {commentUid, before, after});
+    return;
+  }
+
+  // get the source document
+  const {root, space, documentId} = event.params;
+  const documentPath = `${root}/${space}/documents/${documentId}`;
+  const firestore = admin.firestore();
+  logInfo("Firestore document path:", documentPath);
+  const docSnapshot = await firestore.doc(documentPath).get();
+  if (!docSnapshot.exists) {
+    // sanity check
+    logInfo("EARLY EXIT: Document for comment does not exist:", documentPath);
+    return;
+  }
+  const docData = docSnapshot.data();
+
+  if (!docData || !docData.context_id || !docData.uid) {
+    logInfo("EARLY EXIT: Document is missing required fields");
+    return;
+  }
+
+  // FOR NOW: only summarize the cas unit documents
+  // TODO: figure out when to summarize and when to not
+  if (docData.unit !== "cas") {
+    logInfo("EARLY EXIT: document is not a cas unit:", {unit: docData.unit});
+    return;
+  }
+
+  if (["authed", "demo"].includes(root)) {
+    logInfo("EARLY EXIT: Document is in authed or demo root:", {root});
+  }
+  // TODO: like onAnalysisDocumentWritten have two endpoints and then a single handler
+  const prefix = root === "authed" ? `authed/portals/${space}` : `demo/${space}/portals/demo`;
+  // TODO: get the current content of the document from Firebase
+  const firebaseDocumentPath = `${prefix}/classes/${docData.context_id}/users/${docData.uid}/documents/${documentId}`;
+  logInfo("Firebase document path:", firebaseDocumentPath);
+  let content: string|undefined = undefined;
+  try {
+    await (getDatabase().ref(firebaseDocumentPath).once("value", (snapshot) => {
+      content = snapshot.child("content").val() as string;
+    }));
+  } catch (err) {
+    logInfo(`Could not retrieve document ${firebaseDocumentPath}: ${err}`);
+    return;
+  }
```

Comment:
I think the tasks described by the TODO comments here are being done. Isn't that right?